actions/access_keys/get.go

```diff
@@ -9,23 +9,44 @@ import (
 	"github.com/gin-gonic/gin"
 )
 
+// get will get an existing model
+// Get access key godoc
+// @Summary		Get access key - Use (GET) /api/v1/users/current/keys/{id} instead.
+// @Description	This endpoint has been deprecated. Use (GET) /api/v1/users/current/keys/{id} instead.
+// @Tags		Access-key
+// @Produce		json
+// @Param		id query string true "id of the access key"
+// @Success		200	{object} models.AccessKey "AccessKey with given id"
+// @Failure		400	"Bad request - Missing required field: id"
+// @Failure		403	"Forbidden - Access key is not owned by the user"
+// @Failure 	500	"Internal server error - Error while getting access key"
+// @DeprecatedRouter  /v1/access-key [get]
+// @Security	x-auth-xpub
+func (a *Action) oldGet(c *gin.Context) {
+	a.getHelper(c, true)
+}
+
 // get will get an existing model
 // Get access key godoc
 // @Summary		Get access key
 // @Description	Get access key
 // @Tags		Access-key
 // @Produce		json
-// @Param		id query string true "id of the access key"
+// @Param		id path string true "id of the access key"
 // @Success		200	{object} models.AccessKey "AccessKey with given id"
 // @Failure		400	"Bad request - Missing required field: id"
 // @Failure		403	"Forbidden - Access key is not owned by the user"
 // @Failure 	500	"Internal server error - Error while getting access key"
-// @Router		/v1/access-key [get]
+// @Router		/api/v1/users/current/keys/{id} [get]
 // @Security	x-auth-xpub
 func (a *Action) get(c *gin.Context) {
+	a.getHelper(c, false)
+}
+
+func (a *Action) getHelper(c *gin.Context, snakeCase bool) {
 	reqXPubID := c.GetString(auth.ParamXPubHashKey)
 
-	id := c.Query("id")
+	id := c.Params.ByName("id")
```

Comment:
is this going to work for both: old and new endpoint?

Reply:
fixed